src/names.c

```diff
@@ -833,12 +849,17 @@ void stop_name_repair() {
   Rf_errorcall(R_NilValue, "`.name_repair` must be a string or a function. See `?vctrs::vec_as_names`.");
 }
 
-struct name_repair_opts new_name_repair_opts(SEXP name_repair, struct vctrs_arg* arg, bool quiet) {
+struct name_repair_opts new_name_repair_opts(r_obj* name_repair,
+                                             struct vctrs_arg* name_repair_arg,
+                                             bool quiet,
+                                             struct r_lazy call) {
   struct name_repair_opts opts = {
+    .shelter = r_null,
     .type = 0,
-    .fn = R_NilValue,
-    .arg = arg,
-    .quiet = quiet
+    .fn = r_null,
+    .name_repair_arg = name_repair_arg,
+    .quiet = quiet,
+    .call = call
   };
```

Comment:
I guess we just expect that someone else protects the elements of an `struct r_lazy`? Like, it doesn't look like the `shelter` here protects the elements of that `call` object at all.

Reply:
yes the caller protects it. In this case the caller is the R function.